### PR TITLE
add libkvikio wheel build to manifest

### PR DESCRIPTION
**Pull request by msarahan**
https://github.com/rapidsai/kvikio/pull/369 moves the python wheel code one level deeper so that we add support for a dedicated libkvikio wheel. This adjustment needs to be accounted for here. Confused PR run without this change: https://github.com/rapidsai/kvikio/actions/runs/8900188332/job/24441209009#step:7:1446

### CODE REVIEW

**Review by trxcllnt — APPROVED** [review]

**Review by vyasr — CHANGES_REQUESTED** [review]
We also need to bump the feature version here. While we're at it, I would also recommend adding the libkvikio directory since that's happening in the same kvikio PR.

File: features/src/rapids-build-utils/opt/rapids-build-utils/manifest.yaml
Comment (vyasr) [inline features/src/rapids-build-utils/opt/rapids-build-utils/manifest.yaml:48]: We no longer need this flag.
```suggestion
```

**Comment by msarahan** [comment]
I don't think this PR can pass without https://github.com/rapidsai/kvikio/pull/369. @vyasr you wanted to merge this end of the loop first, right? Is it OK to merge with it failing here?

If we want things to be passing prior to merging, then I think we have to approach it on the kvikio side with the part that we ripped out here: https://github.com/rapidsai/kvikio/pull/369/files#diff-1eb4e5fd5611777d4e597ef299a1cb5ba8050c28a2dabbd4fbc56205d69e5ddaR70-R72

then merge this devcontainers PR, then remove the stuff from kvikio.

**Comment by vyasr** [comment]
Yeah, one of these has to be merged while failing. Once the kvikio PR is passing all tests other than the devcontainer jobs, and once you have all the approvals that you need there, we can merge this PR and rerun tests a final time on that repo to make sure everything looks right before merging.

**Comment by msarahan** [comment]
@trxcllnt updated the kvikio path from https://github.com/rapidsai/kvikio/pull/392 in https://github.com/rapidsai/devcontainers/pull/321

This PR now only adds an entry for building the cpp-only shared library wheel, which is utilized in https://github.com/rapidsai/kvikio/pull/369

**Comment by msarahan** [comment]
The kvikio PR has been merged. I'm going to close/reopen this PR to cycle CI and see where we stand.

**Review by vyasr — APPROVED** [review]

**Pull request merged**
